src/qdht.jl

```diff
@@ -0,0 +1,317 @@
+    Y ./= Q.scaleRK
+end
+
+"""
+    *(Q::QDHT, A)
+
+Calculate the forward quasi-discrete Hankel transform of array `A` using the QDHT `Q`.
+
+# Examples
+```jldoctest
+julia> q = QDHT{0, 1}(1e-2, 8); A = exp.(-q.r.^2/(1e-3*q.R));
+julia> q*A
+8-element Array{Float64,1}:
+  4.326937831591551e-6
+  2.3341589529175126e-6
+  7.689558743828849e-7
+  1.546419420523699e-7
+  1.8999259906096856e-8
+  1.4159642663129888e-9
+  7.013670190083954e-11
+ -6.07681871673291e-13
+```
+"""
+function *(Q::QDHT, A)
+    out = similar(A)
+    mul!(out, Q, A)
+end
+
+"""
+    \\(Q::QDHT, A)
+
+Calculate the inverse quasi-discrete Hankel transform of array `A` using the QDHT `Q`.
+
+# Examples
+```jldoctest
+julia> q = QDHT{0, 1}(1e-2, 8); A = exp.(-q.r.^2/(1e-3*q.R));
+julia> Ak = q*A;
+julia> q \\ Ak ≈ A
+true
+```
+"""
+function \(Q::QDHT, A)
+    out = similar(A)
+    ldiv!(out, Q, A)
+end
+
+"""
+    integrateR(A, Q::QDHT; dim=1)
+
+Radial integral of `A`, over the aperture of `Q` in real space.
+
+Assuming `A` contains samples of a function `f(r)` at sample points `Q.r`, then
+`integrateR(A, Q)` approximates ∫f(r)r dr from r=0 to r=∞.
+
+!!! note
+    `integrateR` and `integrateK` fulfill Parseval's theorem, i.e. for some array `A`,
+    `integrateR(abs2.(A), q)` and `integrateK(abs2.(q*A), q)` are equal, **but**
+    `integrateR(A, q)` and `integrateK(q*A, q)` are **not** equal.
+
+!!! warning
+    using `integrateR` to integrate a function (i.e. `A` rather than `abs2(A)`) is only
+    supported for the 0th-order QDHT. For more details see [Derivations](@ref).
+
+# Examples
+```jldoctest
+julia> q = QDHT{0, 1}(10, 128); A = exp.(-q.r.^2/2);
+julia> integrateR(abs2.(A), q) ≈ 0.5 # analytical solution of ∫exp(-r²)r dr from 0 to ∞
+true
+```
+"""
+integrateR(A, Q::QDHT; dim=1) = dimdot(Q.scaleR, A; dim=dim)
+
+"""
+    integrateK(Ak, Q::QDHT; dim=1)
+
+Radial integral of `A`, over the aperture of `Q` in reciprocal space.
+
+Assuming `A` contains samples of a function `f(k)` at sample points `Q.k`, then
+`integrateR(A, Q)` approximates ∫f(k)k dk from k=0 to k=∞.
+
+!!! note
+    `integrateR` and `integrateK` fulfill Parseval's theorem, i.e. for some array `A`,
+    `integrateR(abs2.(A), q)` and `integrateK(abs2.(q*A), q)` are equal, **but**
+    `integrateR(A, q)` and `integrateK(q*A, q)` are **not** equal.
+
+# Examples
+```jldoctest
+julia> q = QDHT{0, 1}(10, 128); A = exp.(-q.r.^2/2);
+julia> integrateR(abs2.(A), q) ≈ 0.5 # analytical solution of ∫exp(-r²)r dr from 0 to ∞
+true
+julia> Ak = q*A;
+julia> integrateK(abs2.(Ak), q) ≈ 0.5 # Same result
+true
+
+```
+"""
+integrateK(Ak, Q::QDHT; dim=1) = dimdot(Q.scaleK, Ak; dim=dim)
+
+"""
+    onaxis(Ak, Q::QDHT; dim=Q.dim)
+
+Calculate on-axis sample in space (i.e. at r=0) from transformed array `Ak`.
+
+# Examples
+```jldoctest
+julia> q = QDHT{0, 1}(10, 128); A = exp.(-q.r.^2/2);
+julia> onaxis(q*A, q) ≈ 1 # should be exp(0) = 1
+true
+```
+"""
+function onaxis(Ak, Q::QDHT{0,n}; dim = Q.dim) where {n}
```

Comment:
Adding an explanatory error may be useful here, e.g.
```julia
onaxis(Ak, Q::QDHT{p, n}; dim=Q.dim) where {p, n} = error("`onaxis` is only supported for 0th-order transforms")
```
Just a `MethodError` might be confusing if the user hasn't read all of the docs.

Reply:
I'd prefer not to for `onaxis`, since in Julia it's typically best to leave unimplemented methods unimplemented rather than implementing them with an error (more on this [here](https://white.ucc.asn.au/2020/04/19/Julia-Antipatterns.html)). And the proposed error does not give any additional information that the `MethodError` doesn't already give:

```julia
julia> q = QDHT{1,1}(1.0, 10);

julia> onaxis(randn(size(q.r)), q)
ERROR: MethodError: no method matching onaxis(::Array{Float64,1}, ::QDHT{1,1,Float64})
Closest candidates are:
  onaxis(::Any, ::QDHT{0,1,T} where T<:Real; dim) at /Users/saxen/Box/projects/Hankel.jl/src/qdht.jl:225
  onaxis(::Any, ::QDHT{0,n,T} where T<:Real; dim) where n at /Users/saxen/Box/projects/Hankel.jl/src/qdht.jl:222
```

The trace shows that `onaxis` is only implemented for zero-order, and we can update the docstring to say why if a user wants to know. For methods we provide that rely on `onaxis`, we can certainly raise a `DomainError`.